libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -681,13 +706,13 @@ public ElasticsearchEntitlementChecker(PolicyManager policyManager) {
         SocketAddress target
     ) {
         if (target instanceof InetSocketAddress isa && isa.getAddress().isMulticastAddress()) {
-            policyManager.checkNetworkAccess(callerClass, NetworkEntitlement.ACCEPT_ACTION | NetworkEntitlement.CONNECT_ACTION);
+            policyManager.checkAllNetworkAccess(callerClass);
         }
```

Comment:
This should probably an if / else for clarity. If all network access was checked successfully, there's no need to do more checks.

Reply:
Yes, it should be an if/else. Fixed!